app/src/main/java/org/fossasia/openevent/app/data/EventRepository.java

```diff
@@ -110,9 +110,21 @@ public EventRepository(IUtilModel utilModel, IDatabaseRepository databaseReposit
 
         Observable<Event> networkObservable = Observable.defer(() ->
             eventService.getEvents(authorization)
-                .doOnNext(events -> databaseRepository.saveList(Event.class, events)
+                .doOnNext(events ->
+                    databaseRepository.saveList(Event.class, events)
                     .subscribe())
-                .flatMapIterable(events -> events));
+                .flatMapIterable(events -> events))
+                .doOnEach(eventNotification -> {
+                    // Download all complete events in one go
+                    if (!eventNotification.isOnNext())
+                        return;
+                    Event event = eventNotification.getValue();
+                    getEvent(event.getId(), false)
```

Comment:
I was saying before calling `getEvent`. But I think its not required now. What do you say?